### PR TITLE
Helm chart: add operator

### PR DESCRIPTION
**Pull request by rmstoi**
Signed-off-by: Ruslan Mstoi <ruslan.mstoi@intel.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,4 +32,5 @@ _build
 _work
 
 *.tgz
-charts/operator/crds
\ No newline at end of file
+charts/operator/crds
+charts/operator/deviceplugin_v1_sgxdeviceplugin.yaml
```
Comment (mythi) [inline .gitignore:36]: how do I modify these parameters using helm?

Comment (mythi) [reply .gitignore:36]: each entry is basically a config entry so all of them should be configurable. I also had a chat with @poussa  and he proposed to make these "included" based on a boolean, e.g. `{{- if .Values.devices.SGX | empty | not }}`...

Comment (rmstoi) [reply .gitignore]: I've pushed a fix to this, to test installation:

`helm install charts/operator  --debug  --dry-run  --generate-name --set sgx.enabled=true`

Comment (mythi) [reply .gitignore]: looks very good to me! one question: can this `<device>.enabled=true` be easily taken to the operators' parameters like this too: https://github.com/intel/intel-device-plugins-for-kubernetes/blob/23eacb3188df5d753792433b73c646611f272d32/deployments/operator/device/sgx/sgx.yaml#L13

Comment (mythi) [reply .gitignore]: @rmstoi this chart does not deploy anything:

```
$ kubectl get pods -n default
No resources found in default namespace.
$ helm status operator-1651210270
NAME: operator-1651210270
LAST DEPLOYED: Fri Apr 29 08:31:10 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing intel-device-plugins-operator.
```

Comment (rmstoi) [reply .gitignore]: > looks very good to me! one question: can this `<device>.enabled=true` be easily taken to the operators' parameters like this too:
> 
> https://github.com/intel/intel-device-plugins-for-kubernetes/blob/23eacb3188df5d753792433b73c646611f272d32/deployments/operator/device/sgx/sgx.yaml#L13

I don't see why not

Comment (rmstoi) [reply .gitignore]: > @rmstoi this chart does not deploy anything:

I pushed a fix to this, with a new templates/operator.yaml originally created by kustomize

File: charts/operator/templates/operator.yaml [hunk 2]
```diff
@@ -0,0 +1,833 @@
...
+      containers:
+      - args:
+        - --metrics-addr=127.0.0.1:8080
+        - --enable-leader-election
+        env:
+        - name: DEVICEPLUGIN_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        image: "{{ .Values.image.hub }}/intel-device-plugins-operator:{{ .Values.image.tag }}"
```
Comment (mythi) [inline charts/operator/templates/operator.yaml:423]:
```suggestion
        image: "{{ .Values.image.hub }}/intel-deviceplugin-operator:{{ .Values.image.tag }}"
```

Comment (mythi) [inline charts/operator/templates/operator.yaml:423]:
```suggestion
        image: "{{ .Values.image.hub }}/intel-deviceplugin-operator:{{ .Values.image.tag }}"
```

**Comment by mythi** [comment]
I installed the operator without SGX and deployed QAT CR but it fails. Looks like the helm chart fails to set cert-manager certs correctly:
```
Error from server (InternalError): error when creating "deployments/operator/samples/deviceplugin_v1_qatde
viceplugin.yaml": Internal error occurred: failed calling webhook "mqatdeviceplugin.kb.io": Post "https://
inteldeviceplugins-webhook-service.device-plugins.svc:443/mutate-deviceplugin-intel-com-v1-qatdeviceplugin
?timeout=10s": x509: certificate is valid for inteldeviceplugins-webhook-service.inteldeviceplugins-system
.svc, inteldeviceplugins-webhook-service.inteldeviceplugins-system.svc.cluster.local, not inteldeviceplugi
ns-webhook-service.device-plugins.svc
```

**Comment by mythi** [comment]
I installed the operator without SGX and deployed QAT CR but it fails. Looks like the helm chart fails to set cert-manager certs correctly:
```
Error from server (InternalError): error when creating "deployments/operator/samples/deviceplugin_v1_qatde
viceplugin.yaml": Internal error occurred: failed calling webhook "mqatdeviceplugin.kb.io": Post "https://
inteldeviceplugins-webhook-service.device-plugins.svc:443/mutate-deviceplugin-intel-com-v1-qatdeviceplugin
?timeout=10s": x509: certificate is valid for inteldeviceplugins-webhook-service.inteldeviceplugins-system
.svc, inteldeviceplugins-webhook-service.inteldeviceplugins-system.svc.cluster.local, not inteldeviceplugi
ns-webhook-service.device-plugins.svc
```

File: charts/operator/templates/operator.yaml [hunk 3]
```diff
@@ -0,0 +1,833 @@
...
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: inteldeviceplugins-serving-cert
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  dnsNames:
+  - inteldeviceplugins-webhook-service.inteldeviceplugins-system.svc
+  - inteldeviceplugins-webhook-service.inteldeviceplugins-system.svc.cluster.local
```
Comment (mythi) [inline charts/operator/templates/operator.yaml:484]:
```suggestion
  - inteldeviceplugins-webhook-service.{{ .Release.Namespace }}svc
  - inteldeviceplugins-webhook-service.{{ .Release.Namespace }}.svc.cluster.local
```

File: charts/operator/templates/operator.yaml [hunk 4]
```diff
@@ -0,0 +1,833 @@
...
+    - UPDATE
+    resources:
+    - pods
+  sideEffects: None
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: inteldeviceplugins-system/inteldeviceplugins-serving-cert
```
Comment (mythi) [inline charts/operator/templates/operator.yaml:691]:
```suggestion
    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/inteldeviceplugins-serving-cert
```

File: charts/operator/templates/operator.yaml [hunk 5]
```diff
@@ -0,0 +1,833 @@
...
+  name: inteldeviceplugins-selfsigned-issuer
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  selfSigned: {}
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: inteldeviceplugins-system/inteldeviceplugins-serving-cert
```
Comment (mythi) [inline charts/operator/templates/operator.yaml:502]:
```suggestion
    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/inteldeviceplugins-serving-cert
```

**Review by mythi — CHANGES_REQUESTED** [review]
some small fixes I needed to make to get this working

**Review by mythi — APPROVED** [review]
thanks!

**Pull request merged**